src/test/java/org/reactome/release/cosmicupdate/COSMICUpdateUtilTest.java

```diff
@@ -0,0 +1,226 @@
+package org.reactome.release.cosmicupdate;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.schema.InvalidAttributeException;
+import org.gk.schema.SchemaClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class COSMICUpdateUtilTest
+{
+
+	private String COSMICFusionExportFile = null;
+	private String COSMICMutationTrackingFile = null;
+	private String COSMICMutantExportFile = null;
+	
+	private static final String COSMIC_GENOMIC_MUTATION_ID = "GENOMIC_MUTATION_ID";
+	private static final String COSMIC_MUTATION_ID = "MUTATION_ID";
+	private static final String COSMIC_LEGACY_MUTATION_ID = "LEGACY_MUTATION_ID";
+
+	private static final String MUTANT_HEADER =  COSMIC_LEGACY_MUTATION_ID + "\t" +  COSMIC_MUTATION_ID+ "\t" + COSMIC_GENOMIC_MUTATION_ID + "\n"; 
+	
+	@Before
+	public void setup() throws IOException
+	{
+		// need to create temp files to read.
+		createTestCosmicFusionExport();
+		createTestCosmicMutantExport();
+		createTestCosmicMutationTracking();
+		
+	}
+
+
+	private void createTestCosmicMutationTracking() throws IOException
+	{
+		Path pathToMutationTracking = Files.createTempFile("mutationTracking", "csv");
+		this.COSMICMutationTrackingFile = pathToMutationTracking.toString();
+		Files.writeString(pathToMutationTracking, MUTANT_HEADER);
+		Files.writeString(pathToMutationTracking, "1234\t1234\tCOSV2323232\n", StandardOpenOption.APPEND);
+		
+	}
+
+
+	private void createTestCosmicMutantExport() throws IOException
+	{
+		Path pathToMutantExport = Files.createTempFile("mutantExport", "csv");
+		this.COSMICMutantExportFile = pathToMutantExport.toString();
+		Files.writeString(pathToMutantExport, MUTANT_HEADER);
+		Files.writeString(pathToMutantExport, "1234\t1234\tCOSV2323232\n", StandardOpenOption.APPEND);
+		
+	}
+
+
+	private void createTestCosmicFusionExport() throws IOException
+	{
+		Path pathToCOSF = Files.createTempFile("COSF", "csv");
+		this.COSMICFusionExportFile = pathToCOSF.toString();
+		// we only use the FUSION_ID field from the COSMIC Fusion Export.
+		Files.writeString(pathToCOSF, "FUSION_ID\n");
+		Files.writeString(pathToCOSF, "1234\n", StandardOpenOption.APPEND);
+		Files.writeString(pathToCOSF, "COSF6321\n", StandardOpenOption.APPEND);
+		Files.writeString(pathToCOSF, "COSF7890\n", StandardOpenOption.APPEND);
+	}
+	
+	
+	@Test
+	public void testValidateAgainstFiles()
```

Comment:
This test looks to be testing many different scenarios.  Could each case get a separate test with common logic in helper methods?

Reply:
What do you mean by "different scenarios"? The different files? I can refactor `validateIdentifiersAgainstFiles` code to call different methods - one for each file, but I think the scope of those methods should be kept private, so the only way to access them is still through `validateIdentifiersAgainstFiles`.

Reply:
I just mean the different cases for the prefixes.  I think this comment was meant to apply to the `testDeterminePrefixes` test method, but I was thinking it would be good to split it into one test method for each kind of possibility - no EWASes, EWASes but with no modified residue, EWASes with modified residues, EWASes with modified residues that are fragment types etc.  

Right now everything is part of the same test and it's very long and hard to understand.